### ISSUE TITLE
can't set root with react native navigation (rn - v 62.2)

### ISSUE THREAD

**Issue by rishav-ss** [magicismight/react-native-root-toast#130]
with react native navigation, android toast is showing inside sidemenu component.
May be default sidemenu is at the top level (as root) so it is showing inside sidemenu from any ither component..
Any Help?

**Comment by sunnylqm** [comment]
https://github.com/magicismight/react-native-root-siblings/issues/52#issuecomment-563252331

**Comment by rishav-ss** [comment]
@sunnylqm  not working, toast still inside sidemenu not on homescreen

**Comment by sunnylqm** [comment]
Can post your code？

**Comment by rishav-ss** [comment]
`let sibling = new RootSiblings(<Container {...props}/>);`
container - 

` <Animated.View style={[Styles.toastView,
        {backgroundColor:background}, isBottom ? Styles.bottom : Styles.top, {borderRadius},
        { transform: [{ translateX: animatedXValue }] }]}>
          { position === "top" && <View style={[Styles.spaceTop]}/>}
         <View style={Styles.innerView}>
           <Text numberOfLines={1} style={[Styles.toastText, textStyle]}>
              {message}
            </Text>
           {Object.keys(button).length > 0 && <Button {...button} />}
         </View>
         { position === "bottom" && <View style={[Styles.spaceBottom]}/>}
      </Animated.View>`

**Comment by rishav-ss** [comment]
this is working on login/signup screen, but after login inside home screen. There is sidemenu/drawer component which is already on Top layer of Home screen.

Whenever I am calling toast in Home screen and sliding a little drawer, toast is showing inside drawer menu not on home

**Comment by sunnylqm** [comment]
Did you see my link above and use RootSiblingParent?

**Comment by rishav-ss** [comment]
Yes I have applied that, still toast is showing inside drawer . I wanted it in Home screen not inside drawer screen

**Comment by rishav-ss** [comment]
For reproducing - 
clone wix/react-ntive-navigation example 
1) call react native root toast on app.js or default screen (It seems to work fine)
2) call root toast on home (any screen with drawer enabled)(It does not work will be shown inside sidemenu)

**Comment by rishav-ss** [comment]
https://github.com/magicismight/react-native-root-toast/pull/93

**Comment by sunnylqm** [comment]
Maybe you need to inject RootSiblingParent into each page components. And maybe build a HOC or base component to reuse the logic.

**Comment by skychx** [comment]
> Maybe you need to inject RootSiblingParent into each page component. And maybe build a HOC or base component to reuse the logic.

Finally I inject RootSiblingParent into each page component to fix this issue in react-native-navigation

**Comment by JesseWeb** [comment]
> > Maybe you need to inject RootSiblingParent into each page component. And maybe build a HOC or base component to reuse the logic.
> 
> Finally I inject `RootSiblingParent` into each page component to fix this issue in react-native-navigation

man that's hardcore for my 100 screens application.

**Comment by skychx** [comment]
> > > Maybe you need to inject RootSiblingParent into each page component. And maybe build a HOC or base component to reuse the logic.
> > 
> > 
> > Finally I inject `RootSiblingParent` into each page component to fix this issue in react-native-navigation
> 
> man that's hardcore for my 100 screens application.

Hi, the method I just described is effective, but not elegant. After our team evaluated it, we simulated `toast` using `Navigation.showOverlay`, we've been live for two months now and it works fine (PS: we have over 200 screens)

**Comment by shadow-boy** [comment]
just use 'react-native-toast-hybrid' to replace the module
